lib/manager/npm/extract/index.ts

```diff
@@ -253,17 +256,28 @@ export async function extractPackageFile(
       return dep;
     }
     const [depNamePart, depRefPart] = hashSplit;
-    const githubOwnerRepo = depNamePart
-      .replace(/^github:/, '')
-      .replace(/^git\+/, '')
-      .replace(/^https:\/\/github\.com\//, '')
```

Comment:
I think the same can be acheived by adding another `replace`:
```js
.replace(/^git@github.com:/, '')
```

Reply:
Theoretically, yes, but I think ideally we would want to arrive at a point where there are a few formats we try to parse separately instead of magical replaces massaging everything to become a string formatted in one way.

Reply:
For debugability and idiomacy.

Reply:
Which change would be more maintainable?

Reply:
I'd argue separate parsers are more maintainable. Though as existing replace stuff seems to work well enough for its cases I don't think there is need to refactor until something breaks or more cases are added.